checkST2110.js

```diff
@@ -70,68 +72,117 @@ a=ts-refclk:ptp=IEEE1588-2008:39-A7-94-FF-FE-07-CB-D0:37
 a=mediaclk:direct=0
 a=mid:secondary`;
 
+
+
+const specExample22 = `v=0 
+o=- 101202 53 IN IP4 10.0.81.54 
+s=237.0.0.50:22000 
+i=Nmos Testing 237.0.0.50:22000 
+t=0 0 
+a=recvonly 
+a=group:DUP PRIMARY SECONDARY 
+m=video 22000 RTP/AVP 98 
+c=IN IP4 237.0.0.50/32 
+a=source-filter: incl IN IP4 237.0.0.50 10.0.81.54 
+a=rtpmap:98 jxsv/90000 
+a=fmtp:98 sampling=YCbCr-4:2:2;width=1280;height=720;packetmode=0;exactframerate=60000/1001;depth=10;TCS=SDR;colorimetry=BT709;SSN=ST2110-22:2019;TP=2110TPN;level=1k-1;sublevel=Sublev3bpp 
+b=AS:116000 
+a=ssrc:0 cname:nmos@nmos.tv 
+a=ts-refclk:ptp=IEEE1588-2008:08-00-11-FF-FE-22-91-3C:127 
+a=mediaclk:direct=0 
+a=mid:PRIMARY m=video 22000 RTP/AVP 98 
+c=IN IP4 237.64.0.50/32 
+a=source-filter: incl IN IP4 237.64.0.50 10.0.81.154 
+a=rtpmap:98 jxsv/90000 
+a=fmtp:98 sampling=YCbCr-4:2:2;width=1280;height=720;packetmode=0;exactframerate=60000/1001;depth=10;TCS=SDR;colorimetry=BT709;SSN=ST2110-22:2019;TP=2110TPN;level=1k-1;sublevel=Sublev3bpp 
+b=AS:116000 a=ssrc:0 cname:nmos@nmos.tv 
+a=ts-refclk:ptp=IEEE1588-2008:08-00-11-FF-FE-22-91-3C:127 
+a=mediaclk:direct=0 
+a=mid:SECONDARY`;
+
+
+
 // Test ST2110-10 Section 7.4 Test 1 - Where mediaclk:direct is used with PTP, offset value is zero
-const test_10_74_1 = sdp => {
+const test_10_74_1 = (sdp, params) => {
   let errors = [];
   let streams = sdp.split(/[\r\n]m=/).slice(1);
-  for ( let s in streams ) {
+  for (let s in streams) {
     let zeroCheck = streams[s].match(mediaclkDirectPattern);
     if (Array.isArray(zeroCheck) && zeroCheck.length > 0 &&
       streams[s].indexOf('IEEE1588-2008') > 0) { // Zero check only PTP clocks
       zeroCheck = zeroCheck.map(z => +(z.trim().split('=')[2]));
-      for ( let x in zeroCheck ) {
+      for (let x in zeroCheck) {
         if (zeroCheck[x] !== 0) {
           errors.push(new Error(`For stream ${s}, the 'mediaclk' attribute shall have a zero offset when direct-referenced PTP timing is in use, as per SMPTE ST 2110-10 Section 7.4.`));
         }
       }
     }
   }
+  if (params.verbose && errors.length == 0)
+    console.log("TEST Passed: Test ST2110-10 Section 7.4 Test 1 - Where mediaclk:direct is used with PTP, offset value is zero");
```

Comment:
I think having a verbose mode is a nice idea, but the duplication of the strings is not good for maintenance. As you've spotted there are several comments that are wrong due to historical copy&paste. I wonder whether we can come up with a pattern that defines a `description` to replace the comment currently usually preceding the function declaration... Then the `if (params.verbose && errors.length == 0)` can be written once, outside all the functions too.

Sketch:
```
const test_10_74_1 = ("Test ST2110-10 Section 7.4 Test 1 - Where mediaclk:direct is used with PTP, offset value is zero",
    (sdp, params) => {
  let errors = [];
  ...
  return errors;
});
```

Potentially bailing out of a test case because `params.should` could be moved out of the tests in the same way, although adding a third member to the tuple might tip us over into using object literals with named fields. Or since there are other reasons that test cases may be skipped, perhaps it is better to keep all the `params` checks within the functions but return something more flexible than the `Error` object, i.e., something that can indicate a test was skipped and why.

Reply:
need to think a bit more about this. I'll check how this syntax works but in general it's a better idea to have the description text as part of the code and reused for printouts than duplicated in the comment.

Reply:
Will track this idea for after this PR.